rundeckapp/grails-spa/packages/ui-trellis/src/app/pages/job/editor/DetailsEditorSection.vue

```diff
@@ -0,0 +1,51 @@
+<template>
+  <details-editor
+    v-if="updatedData"
+    v-model="updatedData"
+    :allow-html="allowHtml"
+  />
+  <json-embed :output-data="updatedData" field-name="jobDetailsJson" />
```

Comment:
this field `jobDetailsJson` still needs to be received on the backend when the form is submitted.  here: https://github.com/rundeck/rundeck/blob/e99997a50d5c17f551014e565b95ad9dd60075a5/rundeckapp/grails-app/services/rundeck/services/ScheduledExecutionService.groovy#L3278-L3278

See example for the options editor, it uses a `jobOptionsJson` field https://github.com/rundeck/rundeck/blob/e99997a50d5c17f551014e565b95ad9dd60075a5/rundeckapp/grails-spa/packages/ui-trellis/src/app/pages/job/editor/OptionsEditorSection.vue#L10-L10 and then uses the value here https://github.com/rundeck/rundeck/blob/e99997a50d5c17f551014e565b95ad9dd60075a5/rundeckapp/grails-app/services/rundeck/services/ScheduledExecutionService.groovy#L3030-L3042

for the `jobDetailsJson`, in the ScheduledExecutionService.jobDefinitionBasic method you will want to simply add the jobDetailsJson data after parsing it into the basicProps data that is used there.

Reply:
I also noticed, that the "jobDetailsJson" field doesn't get correctly updated when the field values are changed.

Reply:
This is something I couldn't make a decision about:

the fields from the details tab were originally submitted as direct inputs. While the json is convenient to load the data, I wasn't sure if should move everything to the json and prevent the inputs from appearing in the form submission, if should keep both json and inputs updated OR if shouldn't use the json at all.

The PR as is, is using the json to load, but updating mainly the inputs only, even though the json is still being submitted. 

What do you recommend?

Reply:
Yes, let's remove the direct input field usage. We want to move to a pure json approach and won't be sending those fields, so we should not use them in the refactored UI.  In later phase we plan to migrate to use API directly, so the json data from this component will be part of that payload.  We will have to make the editor section components return js objects to include in the payload, so we might as well do it now.